TestApp/Search/Sources/SearchViews/SearchFlow.swift

```diff
@@ -0,0 +1,51 @@
+import SwiftUI
+import Navigation
+import Search
+
+public struct SearchFlow: View {
+
+    enum Path: Hashable {
+        case results
+    }
+
+    @StateObject private var path = PushNavigationPath()
+    @Binding private var popToRoot: Bool
+
+    let booking: (_ token: String) -> Void
+
+    public var body: some View {
+        PushNavigationStack(path: path) {
+            search
+        }
+        .popsToRoot(path, trigger: $popToRoot)
+        .destination(for: Path.self) { path in
+            switch path {
+                case .results:
+                    results
+            }
+        }
+        .onDeeplink(SearchDeeplink.self) { _ in
+            // do nothing - default preferences will already pop the stack and switch to this tab
+        }
+        .onDeeplink(SearchResultsDeeplink.self) { _ in
+            path.push(Path.results)
+        }
+    }
+
+    @ViewBuilder var search: some View {
+        SearchScreen {
+            path.push(Path.results)
+        }
+    }
+
+    @ViewBuilder var results: some View {
+        SearchResultsScreen(viewModel: SearchResultsViewModel()) { bookingToken in
```

Comment:
I wonder if initializing VM like this will work fine. It probably needs to be owned by the flow view somehow (`StateObject`). Or is it that the VM is supposed to be lightweight and stateless, so that we do not care that it will be reinitialized?

Reply:
I never saw this type of initializing ObservableObject from the parent, so I am just curious. 

Normally, the `SearchResultsScreen` would instantiate `SearchResultsViewModel()` inside, directly in the @StateObject declaration. Usually, when ObservableObject is passed to subview, it would be an object that is stored somewhere, not re-initialized all the time in the view.

Reply:
How can we test if it works the way we want? 🤔

Reply:
Probably the easiest way would be to change the SearchFlow identity from the outside (changing the `.id()` is the simplest way to force it). This should then make this view initiate `SearchResultsViewModel()` again (which should be what we do not want).

Reply:
You're right, that does trigger the init every time 😞

Reply:
But the same thing happens if I initialize the view model in the view directly...
```swift
@StateObject var viewModel: SearchResultsViewModel = .init()
```
...which is the official way of creating these, isn't it?